File: ALGORITHM/PROGRAMMERS/stackQueue/42583.py

def solution(bridge_length, weight, truck_weights):
    bridge = [0] * bridge_length
    time = 0

    while bridge:
        time += 1
        bridge.pop(0)   # 다리를 건너기

        if truck_weights:
            if truck_weights[0] + sum(bridge) <= weight:
                bridge.append(truck_weights.pop(0))
            else:
                bridge.append(0)

    return time

print(solution(2, 10, [7,4,5,6])) # 8
print(solution(100, 100, [10])) # 101
print(solution(100, 100, [10,10,10,10,10,10,10,10,10,10])) # 110

File: ALGORITHM/PROGRAMMERS/stackQueue/42583.js
function solution(bridge_length, weight, truck_weights) {
    var answer = 0;
    const q = [];
    for (let i = 0; i < bridge_length; i++) {
        q.push(0);
    }
    let w = 0;
    while (truck_weights.length > 0) {
        answer++;
        const out = q.shift();
        w -= out;

        if (w + truck_weights[0] > weight) {
            q.push(0);
        } else {
            const now = truck_weights.shift();
            q.push(now);
            w += now;
        }
    }

    return answer + bridge_length;
}

console.log(solution(2, 10, [7, 4, 5, 6])); // 8
console.log(solution(100, 100, [10])); // 101
console.log(solution(100, 100, [10, 10, 10, 10, 10, 10, 10, 10, 10, 10])); // 110
